lib/trace-object.js

```diff
@@ -0,0 +1,31 @@
+var tracer = require('./tracer');
+var cluster = require('cluster');
+var debug = require('./debug')('trace-object');
+
+module.exports = function sendTraceObject(parentCtl) {
+  if (cluster.isWorker) {
+    tracer().on('trace:object', forwardTraceToMaster);
+    return;
+  }
+
+  if (!parentCtl) {
+    debug('parentCtl not available, all trace objects will be discarded.');
+    return;
+  }
+
+  cluster.on('fork', function(worker) {
+    worker.on('message', function forwardToParentIfTraceObject(msg) {
+      if (msg.cmd !== 'trace:object') return;
+      debug('master received', msg.record);
+      parentCtl.notify(msg);
+    });
+  });
+};
+
+function forwardTraceToMaster(record) {
+  debug('received %j', record);
```

Comment:
record can be quite large, right? many K of data, maybe larger? If it's json, you can write this as

```
debug('received', debug.json(record))
```

See lib/debug.js for definition of debug.json.

If its binary, I suggest truncating it if over a 60 or so bytes long.